apollo-router/src/router_factory.rs

```diff
@@ -483,7 +495,15 @@ pub async fn create_test_service_factory_from_yaml(schema: &str, configuration:
 
     let is_telemetry_disabled = false;
     let service = YamlRouterFactory
-        .create(is_telemetry_disabled, Arc::new(config), schema, None, None)
+        .create(
+            is_telemetry_disabled,
+            Arc::new(config),
+            schema,
+            None,
+            None,
+            // FIXME: default? feels wrong; needs fixture
+            LicenseState::default(),
```

Comment:
__Question__: Does this mean we'll have to reload the Router whenever the license state changes?

And additional question: do we reload the router whenever the `LicenseState` changes?

Reply:
yes and yes!